tests/test_simulate.py

```diff
@@ -161,3 +157,105 @@ def test_snapshot(rng):
         snapshot = json.load(f)
 
     assert s.infections == snapshot
+
+
+class TestResolveDetectionHistory:
+    @pytest.fixture
+    @staticmethod
+    def kwargs():
+        """Baseline keyword arguments for resolve_detection_history tests"""
+        return {
+            "potentially_passive_detected": False,
+            "potentially_active_detected": False,
+            "passive_detection_delay": 5.0,
+            "active_detection_delay": 2.0,
+            "t_exposed": 0.0,
+            "t_recovered": 10.0,
+            "t_infector_detected": None,
+        }
+
+    @staticmethod
+    def f(kwargs: dict):
+        return ringvax.Simulation.resolve_detection_history(**kwargs)
+
+    def test_baseline(self, kwargs):
+        """No potential detections"""
+        assert self.f(kwargs) == {
+            "detected": False,
+            "t_detected": None,
+            "detect_method": None,
+        }
+
+    def test_passive_only(self, kwargs):
+        """Passive detection only"""
+        kwargs["potentially_passive_detected"] = True
+        assert self.f(kwargs) == {
+            "detected": True,
+            "t_detected": 0.0 + 5.0,
```

Comment:
```suggestion
            "t_detected": 5.0,
```

0 + X makes sense to me semantically for active detection, looks a bit weird for passive?